hack/test-cni

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail -x
+
+iidfile=$(mktemp -t docker-iidfile.XXXXXXXXXX)
+
+docker build --iidfile $iidfile --target integration-tests -f ./hack/dockerfiles/test.Dockerfile --force-rm .
+
+iid=$(cat $iidfile)
+rm -f $iidfile
+
+docker run --rm -v /tmp -v /etc/cni/net.d:/etc/cni/net.d -v /opt/cni/bin:/opt/cni/bin --privileged $iid go test ${TESTFLAGS:--v} ${TESTPKGS:-./...}
```

Comment:
Can we make this a "worker" in the integration pkg?

cc @tonistiigi

Reply:
Ideally, yes. I'm fine with the CNI being installed always in the Dockerfile and enabled similarily as we have talked about rootless. We shouldn't need a special Dockerfile